src/components/TimeRanking.tsx

```diff
@@ -1,32 +1,51 @@
 import styled from '@emotion/styled'
 import { useState } from 'react'
 import { ranking } from '@/data/ranking'
+import { useSearchParams } from 'react-router-dom'
+import Spacing from './Spacing'
 
-export default function TimeRanking() {
-  const [selectedGender, setSelectedGender] = useState('ALL')
-  const [selectedRankType, setSelectedRankType] = useState('받고 싶어한')
+const genderOptions = [
+  { label: 'ALL', icon: 'ALL', value: 'ALL' },
+  { label: '여성이', icon: '👩🏻', value: '여성' },
+  { label: '남성이', icon: '👨🏻', value: '남성' },
+  { label: '청소년이', icon: '🧒🏻', value: '청소년' },
+]
+
+const rankTypeTabs = ['받고 싶어한', '많이 선물한', '위시로 받은']
 
-  const filteredRanking = ranking.filter((item) => {
-    return true
-  })
+export default function TimeRanking() {
+  const [searchParams, setSearchParams] = useSearchParams()
+  const [selectedGender, setSelectedGender] = useState(() => searchParams.get('gender') || 'ALL')
+  const [selectedRankType, setSelectedRankType] = useState(() => searchParams.get('rankType') || '받고 싶어한')
+  const [showAll, setShowAll] = useState(false)
+
+  const handleGenderChange = (value: string) => {
+    setSelectedGender(value)
+    searchParams.set('gender', value)
+    setSearchParams(searchParams)
+  }
+
+  const handleRankTypeChange = (value: string) => {
+    setSelectedRankType(value)
+    searchParams.set('rankType', value)
+    setSearchParams(searchParams)
+  }
```

Comment:
함수의 이름은 함수의 내용을 표현해야 한다고 생각해요!
그래서 handle을 붙이기보단 이렇게 함수가 하는 일로 표현해주면 어떨까요?
```suggestion
  const changeGender = (value: string) => {
    setSelectedGender(value)
    searchParams.set('gender', value)
    setSearchParams(searchParams)
  }

  const changeRankType = (value: string) => {
    setSelectedRankType(value)
    searchParams.set('rankType', value)
    setSearchParams(searchParams)
  }
```